pkg/client/cache/reflector.go

```diff
@@ -306,13 +306,34 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	}
 	r.setLastSyncResourceVersion(resourceVersion)
 
+	resyncerrc := make(chan error, 1)
+	go func() {
+		for {
+			select {
+			case <-resyncCh:
+			case <-stopCh:
+				return
+			}
+			if r.canForceResyncNow() {
```

Comment:
This is no longer needed in that case.
If you received sth from resyncCh, that means that you really want to do a resync.

So you should do the resync unconditionally here.